pkg/kubelet/metrics/metrics.go

```diff
@@ -283,3 +352,18 @@ func (pc *podAndContainerCollector) Collect(ch chan<- prometheus.Metric) {
 		prometheus.GaugeValue,
 		float64(runningContainers))
 }
+
+// Helpers for setting the config source metrics. Since counts for these are constant, we only ever set labels.
+
+func SetAssignedNodeConfigSource(val string) {
+	// TODO(mtaufen): will this eliminate previous metric, or generate a parallel one?
+	AssignedNodeConfigSource.WithLabelValues(val).Set(1)
```

Comment:
This approach, implemented like this, is going to result in memory leak, because old values are not GC'd, as you noted in the comment

Reply:
@loburm @x13n FYI, this is the usecase for string metrics